deploy/cloudformation/manifest-pipeline-pipeline.yml

```diff
@@ -125,61 +134,62 @@ Resources:
               - 's3:GetBucketAcl'
               - 's3:PutBucketWebsite'
             Resource:
-              - 'arn:aws:s3:::mellon-manifest-pipeline*'
+              - !Sub 'arn:aws:s3:::${TestStackName}*'
+              - !Sub 'arn:aws:s3:::${ProdStackName}*'
             Effect: 'Allow'
+          # Allow reading from the Pipeline's artifact bucket. Required to deploy the built lambda zips
           - Action:
-              - 's3:GetObject'
-              - 's3:GetObjectVersion'
-              - 's3:GetBucketVersioning'
+              - s3:GetObject
             Resource:
-              - 'arn:aws:s3:::mellon-manifest-pipeline*/*'
-            Effect: 'Allow'
+              - !GetAtt S3Bucket.Arn
+              - Fn::Join:
+                  - ""
+                  -
+                    - !GetAtt S3Bucket.Arn
+                    - "/*"
+            Effect: Allow
+          # Allow the role to read from SSM for the specific template parameters that are needed by manifest-pipeline.yml
           - Action:
               - 'ssm:Get*'
-              - 'ssm:Delete*'
-              - 'ssm:Put*'
-            Resource: !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/all/stacks/mellon*"
+            Resource:
+              - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/all/stacks/${DataBrokerStackName}/*"
+              - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/all/stacks/${ImageServiceTestStackName}/*"
+              - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/all/stacks/${ImageServiceProdStackName}/*"
             Effect: Allow
+          # Allow the role to create SSM resources specified by manifest-pipeline.yml
           - Action:
-              - 's3:PutObject'
+            - 'ssm:Delete*'
```

Comment:
Does CloudFormation need `ssm:Delete*` to tear the stack down? If not, I'd want to consider removing this as there could be data loss in the event of the CloudFormation being changed to remove parameters that are needed elsewhere.

Reply:
I'll double check this one

Reply:
It does, due to it managing these resources https://github.com/ndlib/mellon-blueprints/pull/37/files#diff-3b47693a7c1572e9c25a61f5dc4c097eR40. I spoke to @jonhartzler about this a bit and it sounded like he was going to remove these at some point. But, as long as it is creating these params, it will need to be able to delete them when the stack is deleted.

Reply:
@jonhartzler is this something that you will definitely be removing, or just exploring? I'd like to capture that work.

Reply:
It is something that will be happening.  This will just need get* eventually.